### ISSUE TITLE
Question about specific query 

### ISSUE THREAD

**Issue by dschulz** [GideonLeGrange/mikrotik-java#29]
Hi, 

I'm trying to make a specific query to the router, and it seems that the parser doesn't supports or I'm doing something wrong. Here's an example:

    ip dhcp-server lease print where dynamic=no and active-address in 192.168.88.0/24

This works from the terminal or in a ssh session. The invalid bit seems to be **`and active-address in 192.168.88.0/24`**. 
Is there a way to do this? I skimmed through the API documentation but I still can't figure out. Thanks in advance!

**Comment by GideonLeGrange** [comment]
I can lab this to test it, but before I do that it would be helpful if you can give me a bit more information:
* Can you show me the Java code snippet where you generate the command and call the API?
* Can you show me the stack trace of the error you receive?

**Comment by dschulz** [comment]
Hey, thanks for responding. 
Sure, here's the code I'm using:

        try {
            ApiConnection con = ApiConnection.connectTLS(Config.HOST, ApiConnection.DEFAULT_TLS_PORT, Config.TIMEOUT);
            con.login(Config.USERNAME, Config.PASSWORD);

            List<Map<String, String>> rs1 = con.execute("/ip/dhcp-server/lease/getall where dynamic=no return address,comment");
            rs1.stream().forEach((r) -> {
                System.out.println(r);  
            });

            /*
              Everything is ok here, a list of leases is printed
              ....
              tag=3, data={address=192.168.88.200, comment=Taller}
              tag=3, data={address=192.168.1.103}
              tag=3, data={address=192.168.1.104, comment=fire tv stick}
            */

            List<Map<String, String>> rs2 =  con.execute("/ip/dhcp-server/lease/getall where server=CMTS and address in 192.168.1.0/24");
            
            rs1.stream().forEach((r) -> {
                System.out.println(r);
            });


            con.close();
        } catch { ...


Here's the stacktrace:


    Nov 10, 2015 8:12:47 AM com.dschulz.getleases.GenerarScript main
    SEVERE: null
    me.legrange.mikrotik.impl.ParseException: Expected [=, <, >, !=] but found TEXT at position 62
	    at me.legrange.mikrotik.impl.Parser.expect(Parser.java:183)
	    at me.legrange.mikrotik.impl.Parser.expr(Parser.java:90)
	    at me.legrange.mikrotik.impl.Parser.andExpr(Parser.java:121)
	    at me.legrange.mikrotik.impl.Parser.expr(Parser.java:112)
	    at me.legrange.mikrotik.impl.Parser.where(Parser.java:78)
	    at me.legrange.mikrotik.impl.Parser.parse(Parser.java:27)
	    at me.legrange.mikrotik.impl.Parser.parse(Parser.java:17)
	    at me.legrange.mikrotik.impl.ApiConnectionImpl.execute(ApiConnectionImpl.java:75)
	    at com.dschulz.getleases.GenerarScript.main(GenerarScript.java:43)

**Comment by GideonLeGrange** [comment]
Sorry for the delay. Did you resolve this problem, or do you need assistance? Let me know, if you need help and I will look into it. 

From a quick glance, I don't think "in" is an allowed key word. I'm 99% sure the command language does not support it, but if you disagree, let me know and I'll dig deeper.

**Comment by GideonLeGrange** [comment]
@dschulz I see the command language does support 'in', so apologies for that tangent. It doesn't look (according to the documentation) as if the API language supports "in". I will do a few tests though and see if there is a way of doing it.

**Comment by dschulz** [comment]
Hi Gideon,

Thank you for taking the time to review my question. 
I had to improvise, so what I did in the end was simply querying all the leases and then do the filtering in the application. I used a IP subnet calculator class that popped up clicking "I'm feeling lucky" :-)
I think it would be cool if you find a way to use the 'in' keyword. Unfortunately I have no idea on how to achieve this. I'll keep an eye on your commits and test the library as regularly as I can.
Thanks again!

**Comment by GideonLeGrange** [comment]
@dschulz I went back to the [API "documentation"](http://wiki.mikrotik.com/wiki/Manual:API#Queries) and I experimented a bit, and I think that while the command line has the "in" syntax, the underlying API does not. 

I have posted a [question](http://forum.mikrotik.com/viewtopic.php?f=9&t=103590) on the Mikrotik forum to see if somebody can help me out. 

I'll let you know, but I think for the moment your post-processing solution is most likely the only good option.
